lib/src/server/server.dart

```diff
@@ -102,19 +102,21 @@ class Server {
     server.listen((socket) {
       final connection = ServerTransportConnection.viaSocket(socket);
       _connections.add(connection);
-      ServerHandler_ handler;
+      final handlers = <ServerHandler_>[];
       // TODO(jakobr): Set active state handlers, close connection after idle
       // timeout.
       connection.incomingStreams.listen((stream) {
-        handler = serveStream_(stream);
+        handlers.add(serveStream_(stream));
```

Comment:
We should also remove handlers from the collection. Otherwise we will leak memory.

Reply:
Done

Reply:
That being said, I wonder if me should call `handler.cancel()` in `onError` callback

Reply:
We need to somehow be able to listen to `serveStream_()` being done, and remove the handler from the list there. Maybe that can be done with a StreamController wrapping `stream`. But I think the handler should expose a `.done` or something similar.

Otherwise the `handlers` list will grow without bounds as long as a connection stays open.

We are currently trying to figure out the best way to handle errors in the `onError` callback in https://github.com/grpc/grpc-dart/pull/212 (extracted after some discussion in https://github.com/grpc/grpc-dart/pull/204). I'm still not sure I understand fully what kind of errors can occur in that stream.

Reply:
Humm, it rings me a bell. I already tried to fix a similar problem with #156 (my first attempt had also a leaking issue).

This bug is really annoying because it's easy to crash the server. It would be awesome if you could fix it quickly. I'm going to close this PR and I hope you will come back soon with a fix. Thanks in advance.

Reply:
I'll try to get to it. I cannot promise a particular timeline unfortunately.

Reply:
@sigurdm did you have a chance to look at the issue?